File: app/src/main/java/me/blog/korn123/easyphotomap/thumbnail/AsyncDrawable.java
package me.blog.korn123.easyphotomap.thumbnail;

import android.content.res.Resources;
import android.graphics.Bitmap;
import android.graphics.drawable.BitmapDrawable;

import java.lang.ref.WeakReference;

/**
 * Created by CHO HANJOONG on 2016-08-02.
 */
public class AsyncDrawable extends BitmapDrawable {
    private final WeakReference bitmapWorkerTaskReference;

    public AsyncDrawable(BitmapWorkerTask bitmapWorkerTask, String imagePath) {
//        super(imagePath);
        bitmapWorkerTaskReference = new WeakReference(bitmapWorkerTask);
    }

    public BitmapWorkerTask getBitmapWorkerTask() {
        return (BitmapWorkerTask)bitmapWorkerTaskReference.get();
    }
}

File: app/src/main/java/me/blog/korn123/easyphotomap/thumbnail/AsyncDrawable.kt
package me.blog.korn123.easyphotomap.thumbnail

import android.graphics.drawable.BitmapDrawable
import java.lang.ref.WeakReference

/**
 * Created by CHO HANJOONG on 2016-08-02.
 */
class AsyncDrawable(bitmapWorkerTask: BitmapWorkerTask, imagePath: String) : BitmapDrawable() {
    private val bitmapWorkerTaskReference: WeakReference<*>

    init {
        bitmapWorkerTaskReference = WeakReference(bitmapWorkerTask)
    }

    val bitmapWorkerTask: BitmapWorkerTask
        get() = bitmapWorkerTaskReference.get() as BitmapWorkerTask
}
